hbase-server/src/main/java/org/apache/hadoop/hbase/mob/MobFileCompactionChore.java

```diff
@@ -84,10 +84,10 @@ public MobFileCompactionChore(Configuration conf, int batchSize) {
   protected void chore() {
 
     boolean reported = false;
-
-    try (Connection conn = master.getConnection();
-        Admin admin = conn.getAdmin();) {
-
+    Admin admin = null;
+    try {
+      Connection conn = master.getConnection();
```

Comment:
I feel like we've run into this a few times recently.

Any thoughts about whether it's a good idea to make the ShortCircuitingClusterConnection _not_ close itself (unless we can know that we're in the process of shutting down)?

Reply:
like have it log a debug/info for someone trying to call the normal close but don't do it, then have `inShutdownClose` to actually close? that sounds like a good idea for a follow-on?

Reply:
> that sounds like a good idea for a follow-on?

Fine to think about as follow-on. Just wanted to ask the question while it was fresh :D

Reply:
yes, we should guard-rail the internal connection (in master or RS) from being closed by any shared services by probably wrapping it with SharedConnection which throws an exception when close() is called out (the same way we share it with users through coprocessor ). I'll raise another JIRA for it.

Thanks guys for the ideas and review.